pkg/sharky/recovery.go

```diff
@@ -17,10 +19,13 @@ type Recovery struct {
 	shards []*slots
 }
 
-func NewRecovery(dir string, shardCnt int, shardSize uint32, datasize int) (*Recovery, error) {
+func NewRecovery(dir string, shardCnt int, datasize int) (*Recovery, error) {
 	shards := make([]*slots, shardCnt)
 	for i := 0; i < shardCnt; i++ {
 		file, err := os.OpenFile(path.Join(dir, fmt.Sprintf("shard_%03d", i)), os.O_RDWR|os.O_CREATE, 0666)
+		if errors.Is(err, fs.ErrNotExist) {
```

Comment:
Why is this handled differently? If we dont have `shardCnt` no of shards... we should fail the recovery I feel

Reply:
thanks, this is a good catch, I fixed it to fail if shard is not found, added test.